test/socket.test.js

```diff
@@ -18,61 +13,127 @@ function createClient2() {
     return client;
 }
 
-
-
 describe("on connection", () => {
-    it("should connect client 1", (done) => {
+    it("should connect client ", (done) => {
         client = createClient();
         client.on('connect', () => {
-          
+
             assert(client.connected);
             client.disconnect();
         });
         client.on('disconnect', () => {
             done();
         });
     })
-    it("should connect client 2", (done) => {
-        client2 = createClient2();
-        client2.on('connect', () => {
-           
-            assert(client2.connected);
-            client2.disconnect();
-        });
-        client2.on('disconnect', () => {
-            done();
-        });
-    })
+
 
 
 });
 describe("on reset", () => {
-    it("should reset", (done) => {
-        client = createClient();
-        client.on('_reset', (res) => {
-            assert(res);
-            client.disconnect();
+
+    it("should signin players, create a match and reset data", (done) => {
+        // player1 name
+        const req = {
+            ime: "abc"
+        };
+        // player2 name
+        const req2 = {
+            ime: "def"
+        }
+
+        let countMatches = null;
+
+        // create clients
+        const client = createClient();
+        const client2 = createClient2();
+        const client3 = createClient2();
+
+        //get initial countMatches number for testing
+        client.on("_data", (res) => {
+            countMatches = res.countMatches;
+
```

Comment:
Tukaj lahko testiraš ce je začetno stanje res tako kot bi moralo biti... assert.equal(countMatches,0) ker v idelanih testih moras testirat zacetno stanje in potem ustvaris effekt ki povzroči da se začetno stanje spremeni v končnega.